### PR TITLE
implement exercise 5

### PR DESCRIPTION
**Pull request by steven-mi**
Review: https://github.com/Systems-Development-and-Frameworks/SCP-Foundation/pull/6#pullrequestreview-562703501

### CODE REVIEW

File: backend/src/resolvers.js [hunk 1]
```diff
@@ -1,21 +1,88 @@
...
+            return neo4jgraphql(parent, args, context, resolveInfo);
+        },
+        posts: async (parent, args, context, resolveInfo) => {
+            return neo4jgraphql(parent, args, context, resolveInfo);
+        },
     },
-    User: {
-        posts: (parent, args, context) =>
-            context.dataSources.db.posts.filter((post) => post.author.name === parent.name)
+    Mutation: {
```
Comment (s0557917) [inline backend/src/resolvers.js:15]: ⭐ Your mutations seem to never leave your database in an invalid state

**Review by s0557917 — APPROVED** [review]
Good job! Everything looks good for us.

**Comment by OliKue** [comment]
https://github.com/Systems-Development-and-Frameworks/SCP-Foundation/pull/6#pullrequestreview-562703501

File: backend/README.md [hunk 2]
```diff
@@ -1,13 +1,20 @@
 # backend
 
 ## Project setup
```
Comment (Aimtracker) [inline backend/README.md:3]: ⭐ For choosing a scenario and writing installation instructions in README.md.

File: neo4j/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+.env
```
Comment (Aimtracker) [inline neo4j/.gitignore:1]: ⭐ For not committing sensitive secrets (e.g. API keys) unencrypted to the repository.

File: neo4j/README.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+Create a .env file with your password:
+```
+NEO4J_PASSWORD=test123
+```
+Start docker container:
+```
+docker-compose up
+```
+
+## Why Neo4J with Docker?
```
Comment (Aimtracker) [inline neo4j/README.md:14]: ⭐ For explaining WHY you have chosen the scenario in the README.md.

File: backend/tests/server.spec.js [hunk 5]
```diff
@@ -1,79 +1,93 @@
...
+});
+const bob = new User({
+    name: 'Bob',
+    email: 'bob@example.org',
+    password: '4321'
+});
+
+const bobsPost = new Post({ title: "bobsPost", author: bob })
+
+const cleanDatabase = async() => {
```
Comment (Aimtracker) [inline backend/tests/server.spec.js:28]: ⭐ ⭐ Your software tests are free of side effects.
⭐ Your database is never left in an invalid state.

File: backend/src/driver.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import neo4j from 'neo4j-driver';
+import {NEO4J_HOST, NEO4J_PASSWORD, NEO4J_USERNAME} from "./config";
+
+const driver = neo4j.driver(
```
Comment (Aimtracker) [inline backend/src/driver.js:4]: ⭐ ⭐ Data created in mutations are persisted and your queries and mutations still work as expected.

**Review by Aimtracker — COMMENTED** [review]
Even though our group didn't take the Neo4J approach everything seems to be well implemented and ist readable. You did justify why you chose this approach as well as didn't commit any secrets. Good Job guys!

**Review by medizinmensch — COMMENTED** [review]
Hi abteilung6!

I like your code. Its quite minimal and readable.

One thing regarding your install-instructions: I saw your `neo4j/README` and `neo4j/docker-compose.yml`, but I think others will miss it. Maybe put some instructions in your `/README.md` or at lest your `backend/README.md`. 
But kudos for the extra work 👍 


:star: For choosing a scenario   and writing installation instructions in README.md.
:star:   For explaining WHY you have chosen the scenario in the README.md.
:star: For not committing   sensitive secrets (e.g. API keys) unencrypted to the repository.
:star:   :star: Data created in mutations are persisted and your queries and mutations   still work as expected.
:star: :star: Your software   tests are free of side effects.
:star:   Your database is never left in an invalid state.
:star:   For requesting a review and reviewing another team's PR.

There is sadly one thing missing :x: : 
> :star: You deny the client to   call mutations and queries of your subschema directly.

I left you a comment (see below) explaining where and what is missing. 

That makes **9/10** ⭐   in total.

That's not full points, but its still...

Nice
Youri

File: backend/src/permissions.js [hunk 7]
```diff
@@ -1,14 +1,14 @@
-import {rule, shield} from "graphql-shield"
+import {allow, rule, shield} from "graphql-shield"
```
[the remark this answers is not in the document]
Comment (steven-mi) [reply backend/src/permissions.js:1]: We are not loading the default mutations see: https://github.com/Systems-Development-and-Frameworks/abteilung6/blob/exercise-5/backend/src/neo4j-graphql-js/schema.js#L19

**Pull request merged**